resources/hab_build.rb

```diff
@@ -39,7 +63,10 @@
 
 action :publish do
   execute 'upload-pkg' do
-    command lazy { "#{hab_binary} pkg upload --url #{url} #{hab_studio_path}/src/results/#{artifact}" }
+    command(lazy do
+      url_opt = "--url #{url}" if url
```

Comment:
Reiterating from Slack for posterity: 

The general style is to avoid the use of `new_resource` for properties in custom resources unless the action is using resources that have overlapping properties. For example, we have `cwd` in our `hab_build` resource, so we need to use `new_resource.cwd` in the `execute` resource. Same with `retries` and `environment`.

More information in the [Chef Docs](https://docs.chef.io/custom_resources.html#new-resource-property).